Content.Client/Clothing/ClientClothingSystem.cs

```diff
@@ -61,26 +61,85 @@ public override void Initialize()
     private void OnAppearanceUpdate(EntityUid uid, ClientInventoryComponent component, ref AppearanceChangeEvent args)
     {
         // May need to update jumpsuit stencils if the sex changed. Also required to properly set the stencil on init
-        // when sex is first loaded from the profile.
-        if (!TryComp(uid, out SpriteComponent? sprite) || !sprite.LayerMapTryGet(HumanoidVisualLayers.StencilMask, out var layer))
+        // This is when sex is first loaded from the profile.
+
+        if (!TryComp(uid, out SpriteComponent? sprite) || !TryComp(uid, out HumanoidAppearanceComponent? humanoid))
             return;
 
-        if (!TryComp(uid, out HumanoidAppearanceComponent? humanoid)
-            || humanoid.Sex != Sex.Female
-            || !_inventorySystem.TryGetSlotEntity(uid, "jumpsuit", out var suit, component)
-            || !TryComp(suit, out ClothingComponent? clothing))
+        #region Gendered masks
+        int layer = (int) HumanoidVisualLayers.StencilMask;
+
+        // Set layer to the correct gendered mask layer
+        if (humanoid.Sex == Sex.Male) sprite.LayerMapTryGet(HumanoidVisualLayers.MaleStencilMask, out layer);
+        if (humanoid.Sex == Sex.Female) sprite.LayerMapTryGet(HumanoidVisualLayers.FemaleStencilMask, out layer);
+        if (humanoid.Sex == Sex.Unsexed) sprite.LayerMapTryGet(HumanoidVisualLayers.UnisexStencilMask, out layer);
+
+        if (layer != (int) HumanoidVisualLayers.StencilMask)
         {
-            sprite.LayerSetVisible(layer, false);
-            return;
+            if (!_inventorySystem.TryGetSlotEntity(uid, "jumpsuit", out var suit, component) ||
+                !TryComp(suit, out ClothingComponent? clothing))
+            {
+                sprite.LayerSetVisible(layer, false);
+            }
+            else
+            {
+                // Defaults
+                var mask = clothing.Mask;
+                var prefix = "";
+
+                // Override with gendered mask if it exists
+                if (humanoid.Sex == Sex.Female)
```

Comment:
instead of writing multiple if statements here like above, replace them with one switch block